src/extension.ts

```diff
@@ -145,43 +145,37 @@ export function activate(context: vscode.ExtensionContext) {
             return;
         }
 
-        let bestKey = tryGetCurrentUriKey(map);
-        if (!bestKey) {
+        let jsonPointer = tryGetCurrentJsonPointer(map);
+        if (!jsonPointer) {
             return;
         }
 
-        const shouldOpenDocument = dataPreviewProvider.shouldOpenDocument(map.data, bestKey);
-        const useHtml = dataPreviewProvider.shouldUseHtmlPreview(bestKey);
-        const isShader = dataPreviewProvider.isShader(bestKey);
+        const notDataUri = dataPreviewProvider.uriIfNotDataUri(map.data, jsonPointer);
+        const isShader = dataPreviewProvider.isShader(jsonPointer);
+        const isImage = dataPreviewProvider.isImage(jsonPointer);
+
         let previewUri;
 
-        if (isShader && shouldOpenDocument) {
-            previewUri = Url.resolve(vscode.window.activeTextEditor.document.fileName, shouldOpenDocument);
+        if (!isImage && !isShader) {
+            vscode.window.showErrorMessage('This feature currently works only with images and shaders.');
```

Comment:
Can we allow this to work for accessors now too?  I realize F12 does it, but do we need to prevent it here?  Not a big deal either way.

Reply:
Resolved